core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -620,6 +620,14 @@ private[spark] class TaskSetManager(
     // Note: "result.value()" only deserializes the value when it's called at the first time, so
     // here "result.value()" just returns the value and won't block other threads.
     sched.dagScheduler.taskEnded(tasks(index), Success, result.value(), result.accumUpdates, info)
+    // Kill other task attempts if any as the one attempt succeeded
+    for (attemptInfo <- taskAttempts(index) if attemptInfo.attemptNumber != info.attemptNumber
```

Comment:
Correct the SparkHadoopMapRedUtil.commitTask prevent it from actually committing,  but if that task completion event gets sent back to the driver before your code above sends kill won't it still be marked as success?  Maybe I'm wrong but I seem to remember seeing that happen.  If the commitTask happened at the right time you would see the CommitDenied, but if it happened later it just comes back with success and I don't see how this code helps that.

I agree the second issue, there is actually another jira for it also: https://issues.apache.org/jira/browse/SPARK-10530, but if both are solved by same thing we can just dup them.